rest-app/src/test/java/com/epam/repair/cucumber/CRUDStepDefinitionTestCucumber.java

```diff
@@ -0,0 +1,170 @@
+package com.epam.repair.cucumber;
+
+import com.epam.repair.config.AbstractCucumberSpringConfigurationTest;
+import com.epam.repair.model.RepairOrder;
+import com.epam.repair.service.RepairOrderService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.io.IOException;
+
+import static com.epam.repair.utils.TestUtils.loadTestFile;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class CRUDStepDefinitionTestCucumber extends AbstractCucumberSpringConfigurationTest {
+    int ORDER_ID_1000 = 1000;
+    private String FIRST_NAME_MORE_40 = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+    private String FIRST_NAME_0 = "";
+
+    private ResponseEntity<RepairOrder> responseEntity = null;
+
+    @Autowired
+    private RepairOrderService repairOrderService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+//---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD---ADD
+
+    @Given("A new client brought a device.")
+    public void a_new_customer_brought_a_device() {
+    }
+
+    @When("Add a new client and device.")
+    public void add_a_new_client_and_device() throws IOException {
+        responseEntity = restTemplate.
+                postForEntity(URL, repairOrderAddTest(), RepairOrder.class);
+
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.CREATED);
+    }
+
+    @Then("The order is assigned a unique number.")
+    public void the_order_is_assigned_a_unique_number() {
+
+        assertNotNull(responseEntity.getBody().getRepairOrderId());
+    }
+
+    @Then("Write the client first name more than 40 characters.")
```

Comment:
it might be customized with test parameter and generated in runtime

Reply:
Is regular expression enough?
or should be used:   
| firstName | value |
| firstNamemore40 | AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA |
| firstName0 |  | etc?

Reply:
Remark accepted